


<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="EM based transcripts abundance estimation from nanopore reads mapped to a transcriptome with minimap2">
      
      
        <link rel="canonical" href="https://a-slide.github.io/NanoCount/NanoCount_API_usage/">
      
      
        <meta name="author" content="Adrien Leger">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.1, mkdocs-material-5.1.5">
    
    
      
        <title>Python API - NanoCount</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.df84b5fe.min.css">
      
      
    
    
    
      
    
    
    
      <link rel="stylesheet" href="../css/ansi-colours.css">
    
      <link rel="stylesheet" href="../css/jupyter-cells.css">
    
      <link rel="stylesheet" href="../css/pandas-dataframe.css">
    
    
      
    
    
  </head>
  
  
    <body dir="ltr">
  
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#nanocount-python-api" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid" aria-label="Header">
    <a href="https://a-slide.github.io/NanoCount/" title="NanoCount" class="md-header-nav__button md-logo" aria-label="NanoCount">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header-nav__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header-nav__title" data-md-component="header-title">
      
        <div class="md-header-nav__ellipsis">
          <span class="md-header-nav__topic md-ellipsis">
            NanoCount
          </span>
          <span class="md-header-nav__topic md-ellipsis">
            
              Python API
            
          </span>
        </div>
      
    </div>
    
    
      <div class="md-header-nav__source">
        
<a href="https://github.com/a-slide/NanoCount/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    a-slide/NanoCount
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
        
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://a-slide.github.io/NanoCount/" title="NanoCount" class="md-nav__button md-logo" aria-label="NanoCount">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    NanoCount
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/a-slide/NanoCount/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    a-slide/NanoCount
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../installation/" title="Installation instructions" class="md-nav__link">
      Installation instructions
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" checked>
    
    <label class="md-nav__link" for="nav-3">
      Advanced usage
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M8.59 16.58L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.42z"/></svg>
      </span>
    </label>
    <nav class="md-nav" aria-label="Advanced usage" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        <span class="md-nav__icon md-icon">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
        </span>
        Advanced usage
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../nanocount_io/" title="Input / Output files" class="md-nav__link">
      Input / Output files
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../NanoCount_CLI_usage/" title="Command Line interface" class="md-nav__link">
      Command Line interface
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Python API
        <span class="md-nav__icon md-icon">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 9h14V7H3v2m0 4h14v-2H3v2m0 4h14v-2H3v2m16 0h2v-2h-2v2m0-10v2h2V7h-2m0 6h2v-2h-2v2z"/></svg>
        </span>
      </label>
    
    <a href="./" title="Python API" class="md-nav__link md-nav__link--active">
      Python API
    </a>
    
      
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#activate-virtual-environment" class="md-nav__link">
    Activate virtual environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-nanocount" class="md-nav__link">
    Running NanoCount
  </a>
  
    <nav class="md-nav" aria-label="Running NanoCount">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#basic-command" class="md-nav__link">
    Basic command
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#basic-command-without-file-writing-and-dataframe-output" class="md-nav__link">
    Basic command without file writing and Dataframe output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adding-extra-transcripts-information" class="md-nav__link">
    Adding extra transcripts information
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#relaxing-the-equivalence-threshold" class="md-nav__link">
    Relaxing the equivalence threshold
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nanocount-help" class="md-nav__link">
    NanoCount help
  </a>
  
</li>
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../contributing/" title="Contributing" class="md-nav__link">
      Contributing
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../code_of_conduct/" title="Code of conduct" class="md-nav__link">
      Code of conduct
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../licence/" title="Licence" class="md-nav__link">
      Licence
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../changelog/" title="Changelog" class="md-nav__link">
      Changelog
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#activate-virtual-environment" class="md-nav__link">
    Activate virtual environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-nanocount" class="md-nav__link">
    Running NanoCount
  </a>
  
    <nav class="md-nav" aria-label="Running NanoCount">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#basic-command" class="md-nav__link">
    Basic command
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#basic-command-without-file-writing-and-dataframe-output" class="md-nav__link">
    Basic command without file writing and Dataframe output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adding-extra-transcripts-information" class="md-nav__link">
    Adding extra transcripts information
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#relaxing-the-equivalence-threshold" class="md-nav__link">
    Relaxing the equivalence threshold
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nanocount-help" class="md-nav__link">
    NanoCount help
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/a-slide/NanoCount/edit/master/docs/NanoCount_API_usage.ipynb" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                  
                
                
                <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>

<script>
(function() {
  function addWidgetsRenderer() {
    var mimeElement = document.querySelector('script[type="application/vnd.jupyter.widget-view+json"]');
    var scriptElement = document.createElement('script');
    var widgetRendererSrc = 'https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js';
    var widgetState;

    // Fallback for older version:
    try {
      widgetState = mimeElement && JSON.parse(mimeElement.innerHTML);

      if (widgetState && (widgetState.version_major < 2 || !widgetState.version_major)) {
        widgetRendererSrc = 'jupyter-js-widgets@*/dist/embed.js';
      }
    } catch(e) {}

    scriptElement.src = widgetRendererSrc;
    document.body.appendChild(scriptElement);
  }

  document.addEventListener('DOMContentLoaded', addWidgetsRenderer);
}());
</script>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h1 id="nanocount-python-api">NanoCount python API</h1>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h3 id="activate-virtual-environment">Activate virtual environment</h3>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<pre><code class="python">from NanoCount.NanoCount import NanoCount
from NanoCount.common import jhelp, head
</code></pre>

</div>

</div>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h3 id="running-nanocount">Running NanoCount</h3>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h4 id="basic-command">Basic command</h4>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<pre><code class="python">NanoCount (alignment_file=&quot;./data/aligned_reads.bam&quot;, count_file=&quot;./output/tx_counts.tsv&quot;)
head(&quot;./output/tx_counts.tsv&quot;)
</code></pre>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stderr output_text">
<pre><span class="ansi-blue-intense-fg ansi-bold">## Checking options and input files ##</span>
<span class="ansi-blue-intense-fg ansi-bold">## Initialise Nanocount ##</span>
<span class="ansi-green-fg">    Parse Bam file and filter low quality hits</span>
<span class="ansi-green-fg">    Generate initial read/transcript compatibility index</span>
<span class="ansi-blue-intense-fg ansi-bold">## Start EM abundance estimate ##</span>
    Progress: 3.00 rounds [00:00, 4.59 rounds/s]
<span class="ansi-green-fg">    Exit EM loop after 3 rounds</span>
<span class="ansi-green-fg">    Convergence value: 0.0019235550878568812</span>
<span class="ansi-blue-intense-fg ansi-bold">## Summarize data ##</span>
<span class="ansi-green-fg">    Convert results to dataframe</span>
<span class="ansi-green-fg">    Compute estimated counts and TPM</span>
<span class="ansi-green-fg">    Write file</span>
</pre>
</div>
</div>

<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stdout output_text">
<pre>transcript_name raw                   est_count          tpm                
YHR174W_mRNA    0.5953263925577137    52155.35459919618  595326.3925577137  
YGR192C_mRNA    0.02051266206449663   1797.073298146421  20512.66206449663  
YLR110C_mRNA    0.009451191671993415  827.9999999999991  9451.191671993416  
YOL086C_mRNA    0.00827549995434206   724.9999999999992  8275.49995434206   
YKL060C_mRNA    0.00659757099808236   577.9999999999994  6597.57099808236   
YPR080W_mRNA    0.005222125833257232  457.49999999999955 5222.1258332572315 
YBR118W_mRNA    0.005222125833257232  457.49999999999955 5222.1258332572315 
YLR044C_mRNA    0.0050680303168660346 443.99999999999955 5068.030316866035  
YKL152C_mRNA    0.004166286183910141  364.99999999999966 4166.2861839101415 

</pre>
</div>
</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h4 id="basic-command-without-file-writing-and-dataframe-output">Basic command without file writing and Dataframe output</h4>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
In interactive mode it is also possible not to write the results out but instead to access the data directly as a pandas DataFrame
</div>
</div></p>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<pre><code class="python">nc = NanoCount (alignment_file=&quot;./data/aligned_reads.bam&quot;)
display(nc.count_df)
</code></pre>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stderr output_text">
<pre><span class="ansi-blue-intense-fg ansi-bold">## Checking options and input files ##</span>
<span class="ansi-blue-intense-fg ansi-bold">## Initialise Nanocount ##</span>
<span class="ansi-green-fg">    Parse Bam file and filter low quality hits</span>
<span class="ansi-green-fg">    Generate initial read/transcript compatibility index</span>
<span class="ansi-blue-intense-fg ansi-bold">## Start EM abundance estimate ##</span>
    Progress: 3.00 rounds [00:00, 4.68 rounds/s]
<span class="ansi-green-fg">    Exit EM loop after 3 rounds</span>
<span class="ansi-green-fg">    Convergence value: 0.0019235550878568812</span>
<span class="ansi-blue-intense-fg ansi-bold">## Summarize data ##</span>
<span class="ansi-green-fg">    Convert results to dataframe</span>
<span class="ansi-green-fg">    Compute estimated counts and TPM</span>
</pre>
</div>
</div>

<div class="output_area" markdown="1">

<div class="output_html rendered_html output_subarea ">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>raw</th>
      <th>est_count</th>
      <th>tpm</th>
    </tr>
    <tr>
      <th>transcript_name</th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>YHR174W_mRNA</th>
      <td>5.953264e-01</td>
      <td>52155.354599</td>
      <td>595326.392558</td>
    </tr>
    <tr>
      <th>YGR192C_mRNA</th>
      <td>2.051266e-02</td>
      <td>1797.073298</td>
      <td>20512.662064</td>
    </tr>
    <tr>
      <th>YLR110C_mRNA</th>
      <td>9.451192e-03</td>
      <td>828.000000</td>
      <td>9451.191672</td>
    </tr>
    <tr>
      <th>YOL086C_mRNA</th>
      <td>8.275500e-03</td>
      <td>725.000000</td>
      <td>8275.499954</td>
    </tr>
    <tr>
      <th>YKL060C_mRNA</th>
      <td>6.597571e-03</td>
      <td>578.000000</td>
      <td>6597.570998</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>YHR218W_mRNA</th>
      <td>5.474187e-08</td>
      <td>0.004796</td>
      <td>0.054742</td>
    </tr>
    <tr>
      <th>YBL111C_mRNA</th>
      <td>5.474187e-08</td>
      <td>0.004796</td>
      <td>0.054742</td>
    </tr>
    <tr>
      <th>YOR248W_mRNA</th>
      <td>3.727178e-08</td>
      <td>0.003265</td>
      <td>0.037272</td>
    </tr>
    <tr>
      <th>YOR309C_mRNA</th>
      <td>3.377066e-08</td>
      <td>0.002959</td>
      <td>0.033771</td>
    </tr>
    <tr>
      <th>YLL067C_mRNA</th>
      <td>8.198191e-09</td>
      <td>0.000718</td>
      <td>0.008198</td>
    </tr>
  </tbody>
</table>
<p>3138 rows × 3 columns</p>
</div>
</div>

</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h4 id="adding-extra-transcripts-information">Adding extra transcripts information</h4>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
The <code>extra_tx_info</code> option adds a columns with the transcript lengths and also includes all the zero-coverage transcripts in the results <br />
</div>
</div></p>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<pre><code class="python">nc = NanoCount (alignment_file=&quot;./data/aligned_reads.bam&quot;, extra_tx_info=True)
display(nc.count_df)
</code></pre>

</div>
</div>
</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stderr output_text">
<pre><span class="ansi-blue-intense-fg ansi-bold">## Checking options and input files ##</span>
<span class="ansi-blue-intense-fg ansi-bold">## Initialise Nanocount ##</span>
<span class="ansi-green-fg">    Parse Bam file and filter low quality hits</span>
<span class="ansi-green-fg">    Generate initial read/transcript compatibility index</span>
<span class="ansi-blue-intense-fg ansi-bold">## Start EM abundance estimate ##</span>
    Progress: 3.00 rounds [00:00, 4.34 rounds/s]
<span class="ansi-green-fg">    Exit EM loop after 3 rounds</span>
<span class="ansi-green-fg">    Convergence value: 0.0019235550878568812</span>
<span class="ansi-blue-intense-fg ansi-bold">## Summarize data ##</span>
<span class="ansi-green-fg">    Convert results to dataframe</span>
<span class="ansi-green-fg">    Compute estimated counts and TPM</span>
</pre>
</div>
</div>

<div class="output_area" markdown="1">

<div class="output_html rendered_html output_subarea ">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>raw</th>
      <th>est_count</th>
      <th>tpm</th>
      <th>transcript_length</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>YHR174W_mRNA</th>
      <td>0.595326</td>
      <td>52155.354599</td>
      <td>595326.392558</td>
      <td>1314</td>
    </tr>
    <tr>
      <th>YGR192C_mRNA</th>
      <td>0.020513</td>
      <td>1797.073298</td>
      <td>20512.662064</td>
      <td>999</td>
    </tr>
    <tr>
      <th>YLR110C_mRNA</th>
      <td>0.009451</td>
      <td>828.000000</td>
      <td>9451.191672</td>
      <td>402</td>
    </tr>
    <tr>
      <th>YOL086C_mRNA</th>
      <td>0.008275</td>
      <td>725.000000</td>
      <td>8275.499954</td>
      <td>1047</td>
    </tr>
    <tr>
      <th>YKL060C_mRNA</th>
      <td>0.006598</td>
      <td>578.000000</td>
      <td>6597.570998</td>
      <td>1080</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>YPR200C_mRNA</th>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>393</td>
    </tr>
    <tr>
      <th>YPR201W_mRNA</th>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>1215</td>
    </tr>
    <tr>
      <th>YPR202W_mRNA</th>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>717</td>
    </tr>
    <tr>
      <th>YPR203W_mRNA</th>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>309</td>
    </tr>
    <tr>
      <th>YPR204C-A_mRNA</th>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>483</td>
    </tr>
  </tbody>
</table>
<p>6612 rows × 4 columns</p>
</div>
</div>

</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h4 id="relaxing-the-equivalence-threshold">Relaxing the equivalence threshold</h4>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
The default value is 0.9 (90% of the alignment score of the primary alignment) but this value could be lowered to allow more secondary alignments to be included in the uncertainty calculation.
Lowering the value bellow 0.75 might not be relevant and will considerably increase the computation time.
</div>
</div></p>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<pre><code class="python">NanoCount (alignment_file=&quot;./data/aligned_reads.bam&quot;, count_file=&quot;./output/tx_counts.tsv&quot;, equivalent_threshold=0.8, extra_tx_info=True)
head(&quot;./output/tx_counts.tsv&quot;)
</code></pre>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stderr output_text">
<pre><span class="ansi-blue-intense-fg ansi-bold">## Checking options and input files ##</span>
<span class="ansi-blue-intense-fg ansi-bold">## Initialise Nanocount ##</span>
<span class="ansi-green-fg">    Parse Bam file and filter low quality hits</span>
<span class="ansi-green-fg">    Generate initial read/transcript compatibility index</span>
<span class="ansi-blue-intense-fg ansi-bold">## Start EM abundance estimate ##</span>
    Progress: 17.0 rounds [00:03, 4.44 rounds/s]
<span class="ansi-green-fg">    Exit EM loop after 17 rounds</span>
<span class="ansi-green-fg">    Convergence value: 0.004988077857734706</span>
<span class="ansi-blue-intense-fg ansi-bold">## Summarize data ##</span>
<span class="ansi-green-fg">    Convert results to dataframe</span>
<span class="ansi-green-fg">    Compute estimated counts and TPM</span>
<span class="ansi-green-fg">    Write file</span>
</pre>
</div>
</div>

<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stdout output_text">
<pre>transcript_name raw                  est_count          tpm                transcript_length 
YHR174W_mRNA    0.5847481080119605   51228.61224671184  584748.1080119605  1314              
YGR192C_mRNA    0.015286737423038144 1339.2404921575258 15286.737423038145 999               
YGR254W_mRNA    0.011624369387633806 1018.3877533118225 11624.369387633806 1314              
YLR110C_mRNA    0.00945119167199341  827.9999999999986  9451.19167199341   402               
YJR009C_mRNA    0.009088112600958011 796.1913687447295  9088.112600958011  999               
YOL086C_mRNA    0.008275499954342055 724.9999999999987  8275.499954342056  1047              
YKL060C_mRNA    0.006597570998082356 577.9999999999991  6597.570998082356  1080              
YBR118W_mRNA    0.005222125833257228 457.49999999999926 5222.125833257228  1377              
YPR080W_mRNA    0.005222125833257228 457.49999999999926 5222.125833257228  1377              

</pre>
</div>
</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h3 id="nanocount-help">NanoCount help</h3>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

</div>
</div>
<pre><code class="python">jhelp(NanoCount)
</code></pre>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">

<p><strong>NanoCount</strong> (alignment_file, count_file, min_read_length, min_query_fraction_aligned, equivalent_threshold, scoring_value, convergence_target, max_em_rounds, extra_tx_info, verbose, quiet)</p>
<p>Estimate abundance of transcripts using an EM</p>
<hr>
<ul>
<li><strong>alignment_file</strong> (required) [str]</li>
</ul>
<p>BAM or SAM file containing aligned ONT dRNA-Seq reads including secondary and supplementary alignment</p>
<ul>
<li><strong>count_file</strong> (default: "") [str]</li>
</ul>
<p>Output file path where to write estimated counts (TSV format)</p>
<ul>
<li><strong>min_read_length</strong> (default: 50) [int]</li>
</ul>
<p>Minimal length of the read to be considered valid</p>
<ul>
<li><strong>min_query_fraction_aligned</strong> (default: 0.5) [float]</li>
</ul>
<p>Minimal fraction of the primary hit query aligned to consider the read valid</p>
<ul>
<li><strong>equivalent_threshold</strong> (default: 0.9) [float]</li>
</ul>
<p>Fraction of the alignment score or the alignment length of secondary hits compared to the primary hit to be considered valid hits</p>
<ul>
<li><strong>scoring_value</strong> (default: alignment_score) [str]</li>
</ul>
<p>Value to use for score thresholding of secondary hits either "alignment_score" or "alignment_length"</p>
<ul>
<li><strong>convergence_target</strong> (default: 0.005) [float]</li>
</ul>
<p>Convergence target value of the cummulative difference between abundance values of successive EM round to trigger the end of the EM loop.</p>
<ul>
<li><strong>max_em_rounds</strong> (default: 100) [int]</li>
</ul>
<p>Maximum number of EM rounds before triggering stop</p>
<ul>
<li><strong>extra_tx_info</strong> (default: False) [bool]</li>
</ul>
<p>Add transcripts length and zero coverage transcripts to the output file (required valid bam/sam header)</p>
<ul>
<li><strong>verbose</strong> (default: False) [bool]</li>
</ul>
<p>Increase verbosity for QC and debugging</p>
<ul>
<li><strong>quiet</strong> (default: False) [bool]</li>
</ul>
<p>Reduce verbosity</p>


</div>

</div>
</div>
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid" aria-label="Footer">
        
          <a href="../NanoCount_CLI_usage/" title="Command Line interface" class="md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
            </div>
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Command Line interface
              </div>
            </div>
          </a>
        
        
          <a href="../contributing/" title="Contributing" class="md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Contributing
              </div>
            </div>
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2019 Adrien Leger
          </div>
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/vendor.83fe6e3c.min.js"></script>
      <script src="../assets/javascripts/bundle.7e1cb91c.min.js"></script><script id="__lang" type="application/json">{"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents"}</script>
      
      <script>
        app = initialize({
          base: "..",
          features: [],
          search: Object.assign({
            worker: "../assets/javascripts/worker/search.37585f48.min.js"
          }, typeof search !== "undefined" && search)
        })
      </script>
      
    
  </body>
</html>